apps/storefront/src/components/table/B3PaginationTable.tsx

```diff
@@ -131,24 +138,25 @@ function PaginationTable<GetRequestListParams, ListItem>(
 
   const [isAllSelect, setAllSelect] = useState<boolean>(false);
 
-  const [cacheAllList, setCacheAllList] = useState<Array<CustomFieldItems>>([]);
+  const [cacheAllList, setCacheAllList] = useState<PossibleNodeWrapper<WithRowControls<Row>>[]>([]);
 
-  const [list, setList] = useState<Array<CustomFieldItems>>([]);
+  const [list, setList] = useState<PossibleNodeWrapper<WithRowControls<Row>>[]>([]);
 
   const [selectCheckbox, setSelectCheckbox] = useState<Array<string | number>>([]);
 
   const [isMobile] = useMobile();
 
   const cacheList = useCallback(
-    (edges: Array<CustomFieldItems>) => {
+    (edges: PossibleNodeWrapper<WithRowControls<Row>>[]) => {
       if (!cacheAllList.length) setCacheAllList(edges);
 
       const copyCacheAllList = [...cacheAllList];
 
-      edges.forEach((item: CustomFieldItems) => {
-        const option = item?.node || item;
-        const isExist = cacheAllList.some((cache: CustomFieldItems) => {
-          const cacheOption = cache?.node || cache;
+      edges.forEach((item) => {
+        const option = isNodeWrapper(item) ? item.node : item;
+        const isExist = cacheAllList.some((cache) => {
+          const cacheOption = isNodeWrapper(cache) ? cache.node : cache;
```

Comment:
🤔 instead of a typeguard, should we write some unmarshaller? 

`getWithPossibleNode(cache)`

Reply:
I considered that, but went with this as it's closer to the original code and I wanted to keep this PR as tiny/easy to read as possible

I really wanted to do this with PR with just type definitions and no code changes, but this one forced my hand